src/content/components/FlameChartCanvas.js

```diff
@@ -186,9 +150,77 @@ class FlameChartCanvas extends Component {
     }
   }
 
+  getHoveredStackInfo(
+    {depth, stackTableIndex}: HoveredStackTiming
+  ): string {
+    const { thread, getLabel, stackTimingByDepth } = this.props;
+    const label = getLabel(thread, stackTimingByDepth[depth].stack[stackTableIndex]);
+
+    const duration = stackTimingByDepth[depth].end[stackTableIndex] -
+      stackTimingByDepth[depth].start[stackTableIndex];
+    let durationString;
+    if (duration >= 10) {
+      durationString = duration.toFixed(0);
+    } else if (duration >= 1) {
+      durationString = duration.toFixed(1);
+    } else if (duration >= 0.1) {
+      durationString = duration.toFixed(2);
+    } else {
+      durationString = duration.toFixed(3);
+    }
+
+    return `${durationString}ms - ${label}`;
+  }
+
+  onDoubleClickStack({depth, stackTableIndex}: HoveredStackTiming) {
+    const { stackTimingByDepth, updateProfileSelection } = this.props;
+    updateProfileSelection({
+      hasSelection: true,
+      isModifying: false,
+      selectionStart: stackTimingByDepth[depth].start[stackTableIndex],
+      selectionEnd: stackTimingByDepth[depth].end[stackTableIndex],
+    });
+  }
+
+  hitTest(x: CssPixels, y: CssPixels): HoveredStackTiming | null {
+    const {
+       rangeStart, rangeEnd, viewportLeft, viewportRight, viewportTop,
+       containerWidth, stackTimingByDepth,
+     } = this.props;
+
+    const rangeLength: Milliseconds = rangeEnd - rangeStart;
+    const viewportLength: UnitIntervalOfProfileRange = viewportRight - viewportLeft;
+    const unitIntervalTime: UnitIntervalOfProfileRange = viewportLeft + viewportLength * (x / containerWidth);
+    const time: Milliseconds = rangeStart + unitIntervalTime * rangeLength;
+    const depth = Math.floor((y + viewportTop) / ROW_HEIGHT);
+    const stackTiming = stackTimingByDepth[depth];
+
+    if (!stackTiming) {
+      return null;
+    }
+
+    for (let i = 0; i < stackTiming.length; i++) {
+      const start = stackTiming.start[i];
+      const end = stackTiming.end[i];
+      if (start < time && end > time) {
+        return { depth, stackTableIndex: i };
+      }
+    }
+
+    return null;
+  }
+
+
   render() {
-    this._scheduleDraw();
-    return <canvas className='flameChartCanvas' ref='canvas'/>;
+    const { containerWidth, containerHeight } = this.props;
+
+    return <TimelineCanvas className='flameChartCanvas'
+                           containerWidth={containerWidth}
+                           containerHeight={containerHeight}
+                           onDoubleClickItem={this.onDoubleClickStack}
+                           getHoveredItemInfo={this.getHoveredStackInfo}
+                           drawCanvas={this.drawCanvas}
+                           hitTest={this.hitTest} />;
```

Comment:
Yeah, that would be more consistent with the codebase. I'll update it. I'm not sure (from a bikeshedding point of view) of what utility underscore has for React classes' member functions, as the class instances are never used by anyone, so all class functions are implicitly private.